lib/trie/lookup.go

```diff
@@ -18,24 +18,24 @@ type recorder interface {
 
 // findAndRecord search for a desired key recording all the nodes in the path including the desired node
 func findAndRecord(t *Trie, key []byte, recorder recorder) error {
-	return find(t.root, key, recorder)
+	return find(t.root, key, recorder, true)
 }
 
-func find(parent Node, key []byte, recorder recorder) error {
-	enc, hash, err := parent.EncodeAndHash()
+func find(current Node, key []byte, recorder recorder, isCurrentRoot bool) error {
```

Comment:
why not leave it as `parent`? As in it's `parent` elsewhere mostly.
Although `current` works as well I guess.

Reply:
parent seems rather misleading. parent suggest that there is some other node that is more interesting for this function

Reply:
We are finding a node which can either be the parent itself or one of its children. Why is it misleading?

In some function like `insert` we return a new node given an existing node and a node to insert; how should we call the returned node? I'm happy to do a batch rename if you have better naming. For now I was using `parent` as the argument and `newParent` as the returned node, but maybe `currentNode` and `newNode` would be better? :thinking:

Reply:
changed back to parent